# Tiktokshop

In [32]:
def summarize_api_overview(df):
    """
    Trả về tổng quan về API: số category, số scope, tổng số API
    """
    total_categories = df['category'].nunique()
    total_scopes = df['scope'].nunique()
    total_apis = len(df)
    
    return {
        "total_categories": total_categories,
        "total_scopes": total_scopes,
        "total_apis": total_apis
    }

In [33]:
import json
import pandas as pd

with open("./Document/tiktokapi_summary.json", "r", encoding="utf-8") as f:
    data = json.load(f)

df = pd.DataFrame(data)

df = df[df["scope"].notna()]                     # loại bỏ None
df = df[df["scope"].str.strip() != ""]           # loại bỏ chuỗi rỗng hoặc toàn khoảng trắng

df = df.reset_index(drop=True)

print(f"✅ Số API hợp lệ sau khi loại bỏ scope rỗng: {len(df)}")
display(df.head())
summarize_api_overview(df)


✅ Số API hợp lệ sau khi loại bỏ scope rỗng: 209


title  \
0  Get Authorized Category Assets   
1            Get Authorized Shops   
2               Get Shop Webhooks   
3             Update Shop Webhook   
4             Delete Shop Webhook   

                                         description  \
0  Retrieves the list of business category assets...   
1  Retrieves the list of shops that a seller has ...   
2  Retrieves a shop's webhooks and the correspond...   
3  Updates the shop's webhook URL for a specific ...   
4  Deletes the shop's webhook URL for a specific ...   

                        scope  method                               endpoint  \
0  partner.authorization.info     GET  /authorization/202405/category_assets   
1   seller.authorization.info     GET            /authorization/202309/shops   
2   seller.authorization.info     GET                 /event/202309/webhooks   
3   seller.authorization.info     PUT                 /event/202309/webhooks   
4   seller.authorization.info  DELETE                 /event/202309/webhooks   

        category use_case  
0  Authorization           
1  Authorization           
2          Event           
3          Event           
4          Event

{'total_categories': 18, 'total_scopes': 43, 'total_apis': 209}

In [11]:
import pandas as pd

# Số lượng scope duy nhất
unique_scopes = df['scope'].nunique()
print(f"📊 Tổng số scope duy nhất: {unique_scopes}\n")

# Bước 1: đếm số lần mỗi scope xuất hiện trong từng category
scope_category_counts = df.groupby(['scope', 'category']).size().reset_index(name='Count_in_category')

# Bước 2: tổng số API có scope đó
scope_total_counts = df.groupby('scope').size().reset_index(name='Total_scope_count')

# Bước 3: kết hợp, tính Count_not_in_category = Total_scope_count - Count_in_category
scope_category_stats = pd.merge(scope_category_counts, scope_total_counts, on='scope')
scope_category_stats['Count_not_in_category'] = scope_category_stats['Total_scope_count'] - scope_category_stats['Count_in_category']

# Bước 4: giữ các cột cần thiết
scope_category_stats = scope_category_stats[['scope', 'category', 'Count_in_category', 'Count_not_in_category']]

# === Hiển thị kết quả ===
display(scope_category_stats.sort_values(by='Count_in_category', ascending=False).reset_index(drop=True))


📊 Tổng số scope duy nhất: 43



scope  \
0                                seller.product.basic   
1                            seller.fulfillment.basic   
2                     data.shop_analytics.public.read   
3                          seller.return_refund.basic   
4                 seller.affiliate_collaboration.read   
5                seller.affiliate_collaboration.write   
6                           partner.tap_campaign.read   
7                             seller.customer_service   
8                creator.affiliate_collaboration.read   
9                                seller.product.write   
10                   seller.fulfillment.package.write   
11                                    seller.fbt.info   
12                                seller.finance.info   
13                    seller.affiliate_messages.write   
14                           serller.product.optimize   
15                             seller.promotion.write   
16                seller.global_product.category.info   
17                        seller.global_product.write   
18                              seller.promotion.info   
19                                   seller.logistics   
20                   seller.customer_engagement.write   
21                             creator.showcase.write   
22                         partner.tap_campaign.write   
23                  creator.affiliate.share_link.read   
24                                  seller.order.info   
25                    seller.customer_engagement.read   
26                          seller.authorization.info   
27                                   seller.shop.info   
28                         seller.global_product.info   
29                                   seller.logistics   
30                              seller.product.delete   
31                    seller.creator_marketplace.read   
32                          seller.order.ext_ref.read   
33                          seller.authorization.info   
34                    partner.tap_campaign.link.write   
35                              creator.showcase.read   
36                             creator.affiliate.info   
37                 partner.tap_campaign.product.write   
38                         partner.authorization.info   
39                       seller.global_product.delete   
40                           seller.fulfillment.basic   
41         seller.fulfillment.basic, seller.logistics   
42  seller.fulfillment.basic, seller.return_refund...   
43                       seller.delivery.status.write   
44  seller.order.ext_ref.write.custom, seller.orde...   
45     seller.product.basic, serller.product.optimize   
46                                   seller.shop.info   

                     category  Count_in_category  Count_not_in_category  
0                    Products                 26                      0  
1                 Fulfillment                 13                      1  
2                   Analytics                 11                      0  
3           Return and refund                 11                      0  
4            Affiliate seller                 10                      0  
5            Affiliate seller                 10                      0  
6           Affiliate partner                  9                      0  
7            Customer service                  9                      0  
8           Affiliate creator                  8                      0  
9                    Products                  6                      0  
10                Fulfillment                  6                      0  
11  Fulfilled by TikTok (FBT)                  6                      0  
12                    Finance                  6                      0  
13           Affiliate seller                  6                      0  
14                   Products                  5                      0  
15                  Promotion                  5                      0  
16                   Products                  4     

+ `Count_in_category`: số API có scope đó trong category đó
+ `Count_not_in_category`: số API có scope đó nhưng không thuộc category đó

Báo cáo tổng quan về API

Dưới đây là tổng quan về hệ thống API hiện tại:

| Mục | Số lượng |
|-----|----------|
| Tổng số **Categories** | 18 |
| Tổng số **Scopes** | 43 |
| Tổng số **APIs** | 209 |

## Tiktokshop API Collectors

In [34]:
import json
import pandas as pd

with open("./Document/api_collect_results.json", "r", encoding="utf-8") as f:
    data = json.load(f)

dfTiktok = pd.DataFrame(data)

dfTiktok = dfTiktok.reset_index(drop=True)

print(f"✅ Số API hợp lệ sau khi loại bỏ scope rỗng: {len(dfTiktok)}")
display(dfTiktok.head())
summarize_api_overview(dfTiktok)

✅ Số API hợp lệ sau khi loại bỏ scope rỗng: 117


title  \
0  Check Listing Prerequisites   
1               Get Categories   
2           Get Category Rules   
3               Get Attributes   
4                   Get Brands   

                                         description  category  \
0  Check if a TikTok shop is ready to list produc...  Products   
1  Retrieve the list of product categories availa...  Products   
2  Retrieve the additional requirements (beyond m...  Products   
3  Retrieve the standard built-in product and sal...  Products   
4  Retrieve all available brands for your shop, i...  Products   

                  scope method  \
0  seller.product.basic    GET   
1  seller.product.basic    GET   
2  seller.product.basic    GET   
3  seller.product.basic    GET   
4  seller.product.basic    GET   

                                            endpoint  \
0                      /product/202312/prerequisites   
1                         /product/202309/categories   
2     /product/202309/categories/{category_id}/rules   
3  /product/202309/categories/{category_id}/attri...   
4                             /product/202309/brands   

                                            use_case  
0  Check if a shop meets all requirements before ...  
1  Retrieve the latest list of available product ...  
2  Retrieve specific listing requirements and sup...  
3  Retrieve standard product and sales attributes...  
4  Retrieve all available brands for the shop, in...

{'total_categories': 15, 'total_scopes': 27, 'total_apis': 117}

Báo cáo tổng quan về API

Dưới đây là tổng quan về hệ thống API hiện tại:

| Mục | Số lượng |
|-----|----------|
| Tổng số **Categories** | 15 |
| Tổng số **Scopes** | 27 |
| Tổng số **APIs** | 117 |

In [25]:
import pandas as pd

# Số lượng scope duy nhất
unique_scopes = dfTiktok['scope'].nunique()
print(f"📊 Tổng số scope duy nhất: {unique_scopes}\n")

# Bước 1: đếm số lần mỗi scope xuất hiện trong từng category
scope_category_counts = dfTiktok.groupby(['scope', 'category']).size().reset_index(name='Count_in_category')

# Bước 2: tổng số API có scope đó
scope_total_counts = dfTiktok.groupby('scope').size().reset_index(name='Total_scope_count')

# Bước 3: kết hợp, tính Count_not_in_category = Total_scope_count - Count_in_category
scope_category_stats = pd.merge(scope_category_counts, scope_total_counts, on='scope')
scope_category_stats['Count_not_in_category'] = scope_category_stats['Total_scope_count'] - scope_category_stats['Count_in_category']

# Bước 4: giữ các cột cần thiết
scope_category_stats = scope_category_stats[['scope', 'category', 'Count_in_category', 'Count_not_in_category']]

# === Hiển thị kết quả ===
display(scope_category_stats.sort_values(by='Count_in_category', ascending=False).reset_index(drop=True))

📊 Tổng số scope duy nhất: 27



scope                   category  \
0                             seller.product.basic                   Products   
1                  data.shop_analytics.public.read                  Analytics   
2              seller.affiliate_collaboration.read           Affiliate seller   
3                        partner.tap_campaign.read          Affiliate partner   
4             creator.affiliate_collaboration.read          Affiliate creator   
5                              seller.finance.info                    Finance   
6                                  seller.fbt.info  Fulfilled by TikTok (FBT)   
7                       seller.return_refund.basic          Return and refund   
8                          seller.customer_service           Customer service   
9              seller.global_product.category.info                   Products   
10                        seller.fulfillment.basic                Fulfillment   
11                           seller.promotion.info                  Promotion   
12                        serller.product.optimize                   Products   
13                 seller.affiliate_messages.write           Affiliate seller   
14                 seller.customer_engagement.read        Customer engagement   
15                seller.fulfillment.package.write                Fulfillment   
16                               seller.order.info                     Orders   
17                      seller.global_product.info                   Products   
18                       seller.order.ext_ref.read                     Orders   
19                                seller.logistics                  Logistics   
20                 seller.creator_marketplace.read           Affiliate seller   
21                                seller.logistics                Fulfillment   
22               creator.affiliate.share_link.read          Affiliate creator   
23                           creator.showcase.read          Affiliate creator   
24                          creator.affiliate.info          Affiliate creator   
25                        seller.fulfillment.basic               Supply chain   
26      seller.fulfillment.basic, seller.logistics                  Logistics   
27  seller.product.basic, serller.product.optimize                   Products   
28                                seller.shop.info                  Logistics   

    Count_in_category  Count_not_in_category  
0                  14                      0  
1                  11                      0  
2                  10                      0  
3                   8                      0  
4                   8                      0  
5                   6                      0  
6                   6                      0  
7                   5                      0  
8                   4                      0  
9                   4                      0  
10                  4                      1  
11                  4                      0  
12                  4                      0  
13                  3                      0  
14                  3                      0  
15                  3                      0  
16                  3                      0  
17                  2                      0  
18                  2                      0  
19                  2                      2  
20                  2                      0  
21                  2                      2  
22                  1                      0  
23                  1                      0  
24                  1                      0  
25                  1                      4  
26                  1                      0  
27                  1                      0  
28                  1                      0

In [29]:
def get_apis_by_category(df, category_name):
    scopes = df[df['category'] == category_name]['scope'].unique().tolist()
    return scopes

# Ví dụ:
products_apis = get_apis_by_category(dfTiktok, 'Products')
orders_apis = get_apis_by_category(dfTiktok, 'Orders')

print("📦 APIs cho category Products:")
for api in products_apis:
    print("-", api)

print("\n📝 APIs cho category Orders:")
for api in orders_apis:
    print("-", api)


📦 APIs cho category Products:
- seller.product.basic
- seller.product.basic, serller.product.optimize
- serller.product.optimize
- seller.global_product.category.info
- seller.global_product.info

📝 APIs cho category Orders:
- seller.order.info
- seller.order.ext_ref.read


# Shopee